olp-cpp-sdk-core/src/cache/DefaultCacheImpl.cpp

```diff
@@ -363,13 +356,54 @@ bool DefaultCacheImpl::PromoteKeyLru(const std::string& key) {
   return true;
 }
 
+bool DefaultCacheImpl::PutMutableCache(const std::string& key,
+                                       const leveldb::Slice& value,
+                                       time_t expiry) {
+  if (!mutable_cache_) {
+    return true;
+  }
+
+  // can't put new item if cache is full and eviction disabled
+  const auto item_size = value.size();
+  if (!mutable_cache_lru_ &&
+      mutable_cache_data_size_ + item_size > settings_.max_disk_storage) {
```

Comment:
Might be necessary to also calculate the expiry key size into this, if expiry is set.